src/test/cls_2pc_queue/test_cls_2pc_queue.cc

```diff
@@ -965,33 +965,36 @@ TEST_F(TestCls2PCQueue, MultiProducerConsumer)
 
   const auto max_elements = 128;
   std::vector<std::thread> consumers(max_workers/2);
+  std::mutex consumer_mutex;
   for (auto& c : consumers) {
-    c = std::thread([this, &queue_name, &producer_count] {
+    c = std::thread([this, &queue_name, &producer_count, &retry_happened, &consumer_mutex] {
           librados::ObjectWriteOperation op;
           const std::string marker;
           bool truncated = true;
           std::string end_marker;
           std::vector<cls_queue_entry> entries;
           while (producer_count > 0 || truncated) {
+            if (!retry_happened) {
+              // queue was never full, let it fill
+              std::this_thread::sleep_for(std::chrono::milliseconds(100));
+              continue;
+            }
+            std::lock_guard lock(consumer_mutex);
```

Comment:
why add a mutex over the calls to cls_2pc_queue_list_entries+cls_2pc_queue_remove_entries?

don't we want the consumers to race? if not, maybe just spawn a single consumer thread

Reply:
you are right. no point in testing multiple consumers if we serialize their execution.
in the rgw i use a cls lock to make sure there is no race between consumers.

Reply:
kepts multi threaded readers and a change the deletion to a single thread